src/components/select/__internal__/select-text/select-text.component.js

```diff
@@ -0,0 +1,68 @@
+import React, { useContext } from "react";
+import PropTypes from "prop-types";
+import StyledSelectText from "./select-text.style";
+import useLocale from "../../../../hooks/__internal__/useLocale";
+import { InputContext } from "../../../../__internal__/input-behaviour";
+
+const SelectText = ({
+  disabled,
+  formattedValue,
+  labelId,
+  onClick,
+  onKeyDown,
+  placeholder,
+  readOnly,
+  textId,
+  transparent,
+}) => {
+  const l = useLocale();
+  const inputContext = useContext(InputContext);
+  const hasPlaceholder = !disabled && !readOnly && !formattedValue;
+  const placeholderText = hasPlaceholder
+    ? placeholder || l.select.placeholder()
+    : "";
+
+  return (
+    <StyledSelectText
+      aria-haspopup="listbox"
+      aria-labelledby={labelId}
+      data-element="select-text"
+      disabled={disabled}
+      hasPlaceholder={hasPlaceholder}
+      id={textId}
+      onBlur={inputContext.onBlur}
+      onClick={onClick}
+      onFocus={inputContext.onFocus}
+      onKeyDown={onKeyDown}
+      readOnly={readOnly}
+      role="button"
+      tabIndex="0"
```

Comment:
I think this will create two tab stops with the input and this span, so users will have to tab twice to move past the `Select` when it renders the `SelectText` element

Reply:
input has index of -1